### ISSUE TITLE
Hello, can you export the source file that docker eventually use to compile python？

### ISSUE THREAD

**Issue by dexterhahaha** [beeware/Python-Android-support#18]
Hello, can you export the source file that docker eventually use to compile python？
I am using cmake to build python recently, I hope to refer to what changes have been made to python, the script is not very easy to understand.

**Comment by freakboy3742** [comment]
I'm not sure what you're asking for - or why you're asking for it. 

This project defines a Docker container and build script that compiles Python for use in an Android project. The patches that are applied are all in the `3.7.patches` folder, or are generated by running `3.7.ignore_some_tests.py` over the source tree. You can generate an equivalent source tree by repeating those steps manually.

However, if your aim is to build Python for Android... that's *literally* what this project does, so that you don't have to. What exactly are you trying to achieve?

**Comment by dexterhahaha** [comment]
1. I want to know what has been modified, cpython can be compatible with android.

2. I want to compile cpython through cmake. It is very convenient to use cmake to compile and manage projects.Modifying the source code to compile by patch is not easy to maintain. I need to modify the code of the cpython project and make it compile and run on iOS and Android.

it has some sed command change python source code.
```
# Download & patch Python
ADD downloads/Python-3.7.6.tar.xz .
# Modify ./configure so that, even though this is Linux, it does not append .1.0 to the .so file.
RUN sed -i -e 's,INSTSONAME="$LDLIBRARY".$SOVERSION,,' Python-3.7.6/configure
# Apply a C extensions linker hack; already fixed in Python 3.8+; see https://github.com/python/cpython/commit/254b309c801f82509597e3d7d4be56885ef94c11
RUN sed -i -e s,'libraries or \[\],\["python3.7m"] + libraries if libraries else \["python3.7m"\],' Python-3.7.6/Lib/distutils/extension.py
# Apply a hack to get the NDK library paths into the Python build. TODO(someday): Discuss with e.g. Kivy and see how to remove this.
RUN sed -i -e "s# dirs = \[\]# dirs = \[os.environ.get('SYSROOT_INCLUDE'), os.environ.get('SYSROOT_LIB')\]#" Python-3.7.6/setup.py
# Apply a hack to get the sqlite include path into setup.py. TODO(someday): Discuss with upstream Python if we can use pkg-config for sqlite.
RUN sed -i -E 's,sqlite_inc_paths = [[][]],sqlite_inc_paths = ["/opt/python-build/built/sqlite/include"],' Python-3.7.6/setup.py
# Apply a hack to make platform.py stop looking for a libc version.
RUN sed -i -e "s#Linux#DisabledLinuxCheck#" Python-3.7.6/Lib/platform.py

# Build Python, pre-configuring some values so it doesn't check if those exist.
ENV SYSROOT_LIB=${TOOLCHAIN}/sysroot/usr/lib/${TOOLCHAIN_TRIPLE}/${ANDROID_API_LEVEL}/ \
    SYSROOT_INCLUDE=${NDK}/sysroot/usr/include/
RUN cd Python-3.7.6 && LDFLAGS="$(pkg-config --libs-only-L libffi) $(pkg-config --libs-only-L liblzma) -L${LIBBZ2_INSTALL_DIR}/lib -L$OPENSSL_INSTALL_DIR/lib" \
    CFLAGS="${CFLAGS} -I${LIBBZ2_INSTALL_DIR}/include $(pkg-config --cflags-only-I libffi) $(pkg-config --cflags-only-I liblzma) " \
    ./configure --host "$TOOLCHAIN_TRIPLE" --build "$COMPILER_TRIPLE" --enable-shared \
    --enable-ipv6 ac_cv_file__dev_ptmx=yes \
    --with-openssl=$OPENSSL_INSTALL_DIR \
    ac_cv_file__dev_ptc=no --without-ensurepip ac_cv_little_endian_double=yes \
    --prefix="$PYTHON_INSTALL_DIR" \
    ac_cv_func_setuid=no ac_cv_func_seteuid=no ac_cv_func_setegid=no ac_cv_func_getresuid=no ac_cv_func_setresgid=no ac_cv_func_setgid=no ac_cv_func_sethostname=no ac_cv_func_setresuid=no ac_cv_func_setregid=no ac_cv_func_setreuid=no ac_cv_func_getresgid=no ac_cv_func_setregid=no ac_cv_func_clock_settime=no ac_cv_header_termios_h=no ac_cv_func_sendfile=no ac_cv_header_spawn_h=no ac_cv_func_posix_spawn=no \
    ac_cv_func_setlocale=no ac_cv_working_tzset=no ac_cv_member_struct_tm_tm_zone=no ac_cv_func_sched_setscheduler=no
# Override ./configure results to futher force Python not to use some libc calls that trigger blocked syscalls.
# TODO(someday): See if HAVE_INITGROUPS has another way to disable it.
RUN cd Python-3.7.6 && sed -i -E 's,#define (HAVE_CHROOT|HAVE_SETGROUPS|HAVE_INITGROUPS) 1,,' pyconfig.h
# Adjust timemodule.c to perform data validation for mktime(). The libc call is supposed to do its own
# validation, but on one Android 8.1 device, it doesn't. We leverage the existing AIX-related check in timemodule.c.
RUN cd Python-3.7.6 && sed -i -E 's,#ifdef _AIX,#if defined(_AIX) || defined(__ANDROID__),' Modules/timemodule.c
# Override posixmodule.c assumption that fork & exec exist & work.
RUN cd Python-3.7.6 && sed -i -E 's,#define.*(HAVE_EXECV|HAVE_FORK).*1,,' Modules/posixmodule.c
# Copy libbz2 into the SYSROOT_LIB. This is the IMHO the easiest way for setup.py to find it.
RUN cp "${LIBBZ2_INSTALL_DIR}/lib/libbz2.so" $SYSROOT_LIB
# Compile Python. We can still remove some tests from the test suite before `make install`.
RUN cd Python-3.7.6 && make
```

**Comment by freakboy3742** [comment]
If you want to use Cmake, go right ahead. This project doesn't use CMake, and we have no plans to adopt it. If you want to know what we do to compile... you need to read the code. I don't know what other answer to give you. There's aren't any secrets to what we're doing; it's all documented in code. 

Closing this ticket on the basis that there's nothing actionable.
